### DISCUSSION TITLE
Component closest method

### DISCUSSION THREAD

**Discussion by Baghunts** [GrapesJS/grapesjs#5218] [Show and tell]
For finding some closest element by selector, we can use the `closest` method, which in your turn, uses jQuery's closest method.

jQuery's closest method also returns the element self too. I think it will be great to use the `parents` method instead `closest`, for skipping the element from the search

![image](https://github.com/GrapesJS/grapesjs/assets/133208978/3f1d40ec-6249-47e6-8452-6456dbf2e8e7)

**Comment by bgrand-ch** [comment]
@Baghunts Hello,

Thanks for the highlight.

Will this change improve performance to traverse the DOM?

**Reply by Bagunc** [reply]
Yes, because it will skip the element itself from the search. But here the problem is a more logical case than speed.

For example, we have a structure

`row#mainRow > col > row#innerRow`

If I will use the `closestType("row")` in `#innerRow` component it will return itself because the jQuery `.closest` method starts from elements.
I suggest using jQuery `.parent`, for excluding the element itself in the search.

**Reply by bgrand-ch** [reply]
https://measurethat.net/Benchmarks/Show/17903/0/jquery-parents-vs-closest

`closest` seems to be the fastest 🤔

**Reply by artf** [reply]
Well, honestly I think it's better to keep it as it is even because that might introduce unexpected behavior due to how it works right now.
If you need to skip the component itself just select the parent and then do the closest call (eg. `cmp.parent().closest('...')`)
